src/pages/settings/Payments/TransferBalancePage.js

```diff
@@ -65,65 +65,64 @@ const defaultProps = {
-
-        // Select the default payment account when page is opened,
-        // so that user can see that preselected on choose transfer account page
-        if (!selectedAccount || !selectedAccount.isDefault) {
-            return;
-        }
-
-        PaymentMethods.saveWalletTransferAccountTypeAndID(selectedAccount.accountType, selectedAccount.methodID);
-    }
+function TransferBalancePage(props) {
+    const paymentTypes = [
+        {
+            key: CONST.WALLET.TRANSFER_METHOD_TYPE.INSTANT,
+            title: props.translate('transferAmountPage.instant'),
+            description: props.translate('transferAmountPage.instantSummary', {
+                rate: props.numberFormat(CONST.WALLET.TRANSFER_METHOD_TYPE_FEE.INSTANT.RATE),
+                minAmount: CurrencyUtils.convertToDisplayString(CONST.WALLET.TRANSFER_METHOD_TYPE_FEE.INSTANT.MINIMUM_FEE),
+            }),
+            icon: Expensicons.Bolt,
+            type: CONST.PAYMENT_METHODS.DEBIT_CARD,
+        },
+        {
+            key: CONST.WALLET.TRANSFER_METHOD_TYPE.ACH,
+            title: props.translate('transferAmountPage.ach'),
+            description: props.translate('transferAmountPage.achSummary'),
+            icon: Expensicons.Bank,
+            type: CONST.PAYMENT_METHODS.BANK_ACCOUNT,
+        },
+    ];
 
     /**
      * Get the selected/default payment method account for wallet transfer
      * @returns {Object|undefined}
      */
-    getSelectedPaymentMethodAccount() {
-        const paymentMethods = PaymentUtils.formatPaymentMethods(this.props.bankAccountList, this.props.cardList);
+    const getSelectedPaymentMethodAccount = useCallback(() => {
+        const paymentMethods = PaymentUtils.formatPaymentMethods(props.bankAccountList, props.cardList);
 
         const defaultAccount = _.find(paymentMethods, (method) => method.isDefault);
         const selectedAccount = _.find(
             paymentMethods,
-            (method) => method.accountType === this.props.walletTransfer.selectedAccountType && method.methodID === this.props.walletTransfer.selectedAccountID,
+            (method) => method.accountType === props.walletTransfer.selectedAccountType && method.methodID === props.walletTransfer.selectedAccountID,
         );
         return selectedAccount || defaultAccount;
-    }
+    }, [props.bankAccountList, props.cardList, props.walletTransfer.selectedAccountType, props.walletTransfer.selectedAccountID]);
+
+    useEffect(() => {
+        // Reset to the default account when the page is opened
+        PaymentMethods.resetWalletTransferData();
+    }, []);
+
+    useEffect(() => {
+        const selectedAccount = getSelectedPaymentMethodAccount();
+        if (!selectedAccount || !selectedAccount.isDefault) {
+            return;
+        }
+
+        PaymentMethods.saveWalletTransferAccountTypeAndID(selectedAccount.accountType, selectedAccount.methodID);
+    }, [getSelectedPaymentMethodAccount]);
```

Comment:
This is... interesting... I am sort of wondering why we are running that logic in the constructor before. Because it would only ever run once for the life of the instantiated component and now it will run again whenever the reference changes (which would happen if any props change) 🤔

Not sure how to advise as I'm not sure why we used the constructor in the first place. But my first instinct would be to move this into a `useEffect()` with an empty array as a dependency since that is the closest thing to running logic in the constructor (but isn't exactly the same). 

I haven't run into this much. It seems uncommon to run logic in the constructor unless you are initializing state.

Reply:
Yeah so for context, at first I was trying to use `[]` for the dependencies here to only run the effect once, but I would get the `exhaustive-deps` ESLint warning. Same thing if I didn't `useCallback()` for `getSelectedPaymentMethodAccount`.

I'll go ahead and try your suggestion though, thanks.